### PR TITLE
SnapKit #2 - UserProfileView

### PR DESCRIPTION
**Pull request by matej494**
Predecessor - MovieDetailsVIew

### CODE REVIEW

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 1]
```diff
@@ -15,9 +15,18 @@ class UserProfileViewController: UIViewController {
     private let userMovieIDs = SavedMoviesManager()
     private lazy var movieIDs = userMovieIDs.loadUserMovieIDs()
     private lazy var movies = data.loadMovies(with: movieIDs)
+    private let flowLayout = UICollectionViewFlowLayout()
+    private lazy var collectionView = UICollectionView(frame: CGRect.zero, collectionViewLayout: flowLayout).autolayoutView()
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        self.view.backgroundColor = .white
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Redunant `self`.

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 2]
```diff
@@ -15,9 +15,18 @@ class UserProfileViewController: UIViewController {
...
+
+        setupCollectionView()
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.register(MovieCollectionViewCell.self, forCellWithReuseIdentifier: "cell")
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Why don't we move these three lines of customisation code into the `setupCollectionView`?

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 3]
```diff
@@ -57,8 +66,28 @@ extension UserProfileViewController: UICollectionViewDelegate {
...
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let size = (collectionView.bounds.size.width - 30) / 2
+        return CGSize(width: size, height: size)
+    }
+}
+
 // MARK: - Private Methods Extension
 private extension UserProfileViewController {
+    func setupCollectionView() {
+        self.view.addSubview(collectionView)
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Redunant `self`.

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 4]
```diff
@@ -57,8 +66,28 @@ extension UserProfileViewController: UICollectionViewDelegate {
...
+        flowLayout.scrollDirection = .horizontal
+        flowLayout.minimumInteritemSpacing = 10
+        flowLayout.sectionInset = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+        collectionView.backgroundColor = .white
+        let sizeOfCell = (self.view.bounds.size.width - 30) / 2
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Redunant `self`.

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 5]
```diff
@@ -75,16 +67,17 @@ extension UserProfileViewController: UICollectionViewDelegateFlowLayout {
...
-            $0.height.equalTo(sizeOfCell + 20)
+    func setupView() {
+        navigationItem.title = LocalizationKey.UserProfile.navigationBarTitle.localized()
+        navigationItem.largeTitleDisplayMode = .always
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .stop, target: self, action: #selector(dismissViewController))
+        view.backgroundColor = .white
+        view.addSubview(userProfileView)
+        userProfileView.collectionView.delegate = self
+        userProfileView.collectionView.dataSource = self
+        userProfileView.collectionView.register(MovieCollectionViewCell.self, forCellWithReuseIdentifier: "cell")
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Can we move this into the `UserProfileView`? This is not anything specific to the `view controller`, but rather to the `view` itself.

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 6]
```diff
@@ -6,21 +6,19 @@
...
+import SnapKit
 import RealmSwift
 
 class UserProfileViewController: UIViewController {
-    private let reuseIdentifier = "cell"
     private let data = DataController()
     private let userMovieIDs = SavedMoviesManager()
     private lazy var movieIDs = userMovieIDs.loadUserMovieIDs()
     private lazy var movies = data.loadMovies(with: movieIDs)
+    private let userProfileView = UserProfileView().autolayoutView()
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: Lets use the `static` `autolayoutView` method from the `UIView`.

File: Movieator/App/User profile/UserProfileViewController.swift [hunk 7]
```diff
@@ -6,21 +6,19 @@
...
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        navigationItem.title = LocalizationKey.MovieList.navigationBarTitle.localized()
-        navigationItem.largeTitleDisplayMode = .always
-        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .stop, target: self, action: #selector(dismissViewController))
+        setupView()
```
Comment (dkulundzic) [inline Movieator/App/User profile/UserProfileViewController.swift]: We should override the initialiser and invoke `setupView` from there, so we're sure that, once initialised, our `view controller` is fully ready.

**Comment by matej494** [comment]
Successfully merged branch `user-profile-view`into branch `develop` after resolving conflicts.

**Pull request merged**